yt_dlp/extractor/gedidigital.py

```diff
@@ -1,123 +1,279 @@
+    def _real_extract(self, url):
+        url, media_type, media_id, track_id = self._match_valid_url(url).group(
+            'url', 'type', 'id', 'trtId')
+        self.IE_NAME = f'gedi:{media_type}'
+
+        webpage = self._download_webpage(url, media_id)
+
+        formats = []
+        if media_type == 'video':
+            media_data = self._search_json(
+                r'BrightcoveVideoPlayerOptions\s*=', webpage, 'media_data', media_id,
+                transform_source=lambda s: js_to_json(s.replace('\'[{', '[{').replace('}]\'', '}]')))
+            for fmt in media_data.get('videoSrc'):
+                if 'src' not in fmt or 'type' not in fmt:
+                    continue
+                ext = mimetype2ext(fmt['type'])
+                if ext == 'm3u8':
+                    formats.extend(self._extract_m3u8_formats(fmt['src'], media_id, m3u8_id='hls'))
+                elif ext == 'mp4':
+                    vbr = int_or_none(self._search_regex(
+                        r'video-rrtv-(\d+)', fmt['src'], 'vbr', default=None))
+                    formats.append({
+                        'format_id': join_nonempty('mp4', vbr),
+                        'url': fmt['src'],
+                        'vbr': vbr,
+                    })
+
+        if media_type in ('audio', 'playlist', 'podcast'):
+            # update the media_id (needed for podcasts)
+            media_id = self._search_regex(r'data=[\'"]audioSource(\d+)[\'"]', webpage, 'media_id', default=media_id)
+
+            media_data = self._search_json(
+                rf'audioSource{media_id}\s*=', webpage, 'media_data', media_id,
+                contains_pattern=r'\[(?s:.+)\];', transform_source=js_to_json)
+            if media_type == 'audio':
+                track_id = traverse_obj(media_data, (0, 'trtId'))
+
+            playlist_entries = []
+            for episode in media_data:
+                if track_id == episode['trtId']:
+                    if 'm3u8' == determine_ext(episode.get('audio_url')):
+                        formats.extend(self._extract_m3u8_formats(
+                            episode['audio_url'], media_id, m3u8_id='audio-hls'))
+                    media_data = episode
+                    break
+                elif not track_id:
+                    # if no specific episode is selected return a playlist
+                    playlist_entries.append(self.url_result(f'{url}#{episode["trtId"]}'))
+
+            if playlist_entries:
+                return self.playlist_result(
+                    playlist_entries, playlist_id=media_id,
+                    playlist_title=self._generic_title(url, webpage))
+
+        extra_data = self._search_json_ld(webpage, media_id, default={})
+
+        return {
+            'id': track_id or media_id,
+            'title': (media_data.get('videoTitle')
+                      or media_data.get('title')
+                      or extra_data.get('title')),
```

Comment:
```suggestion
            'title': traverse_obj(media_data, 'videoTitle', 'title'),
```
same below

(see comment about json_ld)